src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs

```diff
@@ -2372,32 +2378,24 @@ public void SwitchExpression02()
                             N(SyntaxKind.IdentifierToken, "b");
                         }
                         N(SyntaxKind.ColonToken);
-                        N(SyntaxKind.SimpleLambdaExpression);
+                        N(SyntaxKind.IdentifierName);
                         {
-                            N(SyntaxKind.Parameter);
-                            {
-                                N(SyntaxKind.IdentifierToken, "c");
-                            }
-                            N(SyntaxKind.EqualsGreaterThanToken);
-                            N(SyntaxKind.IdentifierName);
-                            {
-                                N(SyntaxKind.IdentifierToken, "d");
-                            }
+                            N(SyntaxKind.IdentifierToken, "c");
                         }
                     }
                 }
-                N(SyntaxKind.CloseBraceToken);
+                M(SyntaxKind.CloseBraceToken);
             }
             EOF();
         }
```

Comment:
>} [](start = 8, length = 1)

Perhaps test `switch { a when b ? c : d : e }`.

Reply:
Yes; adding a test.

---
In reply to: [185264756](https://github.com/dotnet/roslyn/pull/26524#discussion_r185264756) [](ancestors = 185264756)